src/config/middleware/experimental-metrics-csv-middleware.ts

```diff
@@ -0,0 +1,164 @@
+import {Middleware, MiddlewareRequestHandler} from './middleware';
+import {Metadata, StatusObject} from '@grpc/grpc-js';
+import {Message} from 'google-protobuf';
+import * as fs from 'fs';
+import {MomentoLogger, MomentoLoggerFactory} from '../logging/momento-logger';
+
+function headerFields(): Array<string> {
+  return [
+    'numActiveRequests',
+    'requestType',
+    'status',
+    'startTime',
+    'endTime',
+    'duration',
+    'requestSize',
+    'responseSize',
+  ];
+}
+
+interface RequestMetrics {
+  numActiveRequests: number;
+  requestType: string;
+  status: number;
+  startTime: number;
```

Comment:
Thinking ahead: there are a few timestamps we will care about. (1) when the client request was initiated; (2) when the grpc request is scheduled; (3) when the grpc/http request actually goes out.

There could be delays between (1) and (2) because of CPU contention; and there could be delays between (2) and (3) because of too many concurrent http streams.

It looks like this will cover (1)? Maybe (2) if the delta between (1) and (2) isn't a big deal? Just wondering if we have any way to tell when the request actually goes out vs when it is scheduled.

Reply:
check out the `on*` callbacks that are defined [here](https://github.com/momentohq/client-sdk-nodejs/pull/300/files#diff-a5d135b804012d314a7ce8a6b8c5627c5fb8825c63aad84e7fd97fb0e651e9f8R4).  Those are the hooks that I have available from the Interceptor AFAIK.

I could theoretically add a timestamp that gets captured in `onRequestBody`, but I don't know if we have any guarantee of what time that happens out of (1), (2), and (3) on your list, and I don't think I have any other hooks.

That said, if you think there's a chance that timestamp might be valuable it's easy enough to add.  LMK if you have a suggestion for a name :)  `requestBodyTime`?

Reply:
No suggestions on a name 😬